### PR TITLE
Cinder

### PR DESCRIPTION
**Pull request by hemna**
restructured the code a bit to be a pbr based app, should tag the repo for a version number.
Added the cinderbackend exporter to fetch all cinder backends/pools and report stats about
total capacity, free capacity, allocated capacity, overcommit ratio, max overcommit ratio, 
free until overcommit

### CODE REVIEW

File: openstack_exporter/exporter.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+@click.option("--port", metavar="<port>", default=9102,
+              help="specify exporter serving port")
+@click.option("-d", "--debug", metavar="<debug>", is_flag=True,
+              default=False, help="enable debug")
+@click.option("-c", "--config", metavar="<config>",
+              help="path to rest config")
+@click.option("-u", "--user", metavar="<user>",
+              help="user used with master password")
+@click.option("-p", "--password", metavar="<password>",
+              help="specify password to log in")
```
Comment (viennaa) [inline openstack_exporter/exporter.py]: Not handled in Code, either pass them through or remove (from main() too) pls.

File: openstack_exporter/exporter.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    else:
+        logging.error("Config file doesn't exist: " + config_file)
+        exit(0)
+
+
+@click.command()
+@click.option("--port", metavar="<port>", default=9102,
+              help="specify exporter serving port")
+@click.option("-d", "--debug", metavar="<debug>", is_flag=True,
+              default=False, help="enable debug")
```
Comment (viennaa) [inline openstack_exporter/exporter.py]: Debug does not switch anything. Handled by config file

File: openstack_exporter/collectors/cinderbackend.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+class CinderBackendCollector(BaseCollector.BaseCollector):
+
+    def describe(self):
+        yield GaugeMetricFamily('cinder_total_capacity',
+                                'Cinder total capacity in GiB')
+        yield GaugeMetricFamily('cinder_free_capacity',
+                                'Cinder free capacity in GiB')
+        yield GaugeMetricFamily('cinder_allocated_capacity',
+                                'Cinder allocated capacity in GiB')
```
Comment (viennaa) [inline openstack_exporter/collectors/cinderbackend.py]: Would rename metrics to have unit in metric name straight:
`cinter_total_capacity_gib` + rest

File: openstack_exporter/collectors/cinderbackend.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        yield GaugeMetricFamily('cinder_max_oversubscription_ratio',
+                                'Cinder max overcommit ratio')
+        yield GaugeMetricFamily('cinder_overcommit_ratio',
+                                'Cinder Overcommit ratio')
+        yield GaugeMetricFamily('cinder_free_until_overcommit',
```
Comment (viennaa) [inline openstack_exporter/collectors/cinderbackend.py]: Do we have a unit here too?

**Review by viennaa — CHANGES_REQUESTED** [review]
Pretty solid and very fast. Some minor changes in regards to params and metric naming and we are good to go.

File: openstack_exporter/collectors/cinderbackend.py (hunk 4 above)
Comment (grandchild) [reply openstack_exporter/collectors/cinderbackend.py:36]: I think ratios implicitly have a unit of 1 (a.k.a. no unit) and also conveniently describes the possible range, [0-1], so I think this is good as it is.

I agree on the units for the *_gib metrics.

File: openstack_exporter/exporter.py (hunk 1 above)
Comment (hemna) [reply openstack_exporter/exporter.py]: removed

File: openstack_exporter/exporter.py (hunk 2 above)
Comment (hemna) [reply openstack_exporter/exporter.py]: removed

File: openstack_exporter/collectors/cinderbackend.py (hunk 3 above)
Comment (hemna) [reply openstack_exporter/collectors/cinderbackend.py]: added

File: openstack_exporter/collectors/cinderbackend.py (hunk 4 above)
Comment (hemna) [reply openstack_exporter/collectors/cinderbackend.py]: added gib

**Review by viennaa — APPROVED** [review]
Awesome, let's go for it.

**Pull request merged**